webapp/src/components/dataset_input/FileSelection.vue

```diff
@@ -0,0 +1,171 @@
+        class="border-dashed rounded-xl border-disco-cyan flex flex-col justify-center items-center min-h-48"
+        :class="
+          isDragHoverActive ? 'bg-blue-100 opacity-75 border-8' : 'border-2'
+        "
+        @dragenter="onDragEnter"
+        @dragleave="onDragLeave"
+        @drop="async (e: DragEvent) => await dragFiles(e)"
+      >
+        <p class="p-4 text-lg text-disco-blue flex-wrap justify-center">
+          <span>Drag and drop the {{ fileType }} or</span>
+        </p>
+        <label class="mb-6">
+          <span
+            class="px-4 py-2 min-w-[8rem] text-lg uppercase text-white bg-disco-cyan rounded duration-200 hover:bg-white hover:outline hover:outline-disco-cyan hover:outline-2 hover:text-disco-cyan hover:cursor-pointer"
+          >
+            select {{ fileType }}
+          </span>
+          <input
+            ref="inputFileElement"
+            type="file"
+            :multiple="multiple"
+            :accept="acceptFilter"
+            class="hidden"
+            @change="async () => await submitFiles()"
+          />
+        </label>
+      </div>
+      <!-- Display what has been connected -->
+      <div
+        v-if="model === undefined"
+        class="flex justify-end items-center mt-1"
+      >
+        <span
+          class="hover:cursor-pointer"
+          v-tippy="{
+            content: 'Data always stays on your device and is never shared.',
+          }"
+        >
+          <i class="fa fa-lock mr-1" />
+        </span>
+      </div>
+
+      <!-- Display some text if specified -->
+      <div
+        v-if="$slots.default && model === undefined"
+        class="flex justify-center mt-5"
+      >
+        <p class="text-slate-500 text-sm">
+          <span><slot /></span>
+        </p>
+      </div>
+
+      <!-- If only one file is connected, display its name, if multiple display the number of files -->
+      <div
+        v-if="model !== undefined"
+        class="pt-4 flex flex-col items-center pb-5"
+      >
+        <div
+          class="mb-4 flex justify-center items-center text-center md:text-left sm:text-lg text-disco-blue"
+        >
+          <span v-if="multiple"
+            >Number of selected files:
+            <span class="pl-1 text-xl">{{ model.size }}</span></span
+          >
+          <span v-else class="pl-1">{{ model.first()?.name ?? "none" }}</span>
+        </div>
+        <div>
+          <CustomButton @click="clearFiles">
+            clear {{ fileType }}
+          </CustomButton>
+        </div>
+      </div>
+    </section>
+  </article>
+</template>
+
+<script lang="ts" setup>
+import { Set } from "immutable";
+import { ref, computed } from "vue";
+
+import CustomButton from "@/components/simple/CustomButton.vue";
+
+const props = withDefaults(
+  defineProps<{
+    type: "tabular" | "image" | "text";
+    multiple?: boolean; // accept one or multiple files
+  }>(),
+  {
+    multiple: false,
+  },
+);
+
+const model = defineModel<Set<File>>();
```

Comment:
I think using the name `model` can be a bit confusing with an ML model. I understand it references the vus.js v-model but I don't think it is necessary to include it in the name. Something like `files` would make more sense to me.